firmware/thruna/BMS/Inc/App/App_Soc.h

```diff
@@ -37,3 +71,70 @@
  *                                not between 0 and 100 inclusive
  */
 ExitCode App_Soc_Vote(float max_abs_difference, float soc_1, float soc_2, float soc_3, float *result);
+
+/**
+ * Create the SocStats object
+ * @return struct SocStats pointer to object
+ */
+struct SocStats *App_SocStats_Create(float initial_charge_value, uint16_t soc_address, struct Accumulator *accumulator);
+
+/**
+ * Destroy the SocStats object
+ */
+void App_SocStats_Destroy(struct SocStats *soc_stats);
+
+/**
+ * Return the active address on the EEPROM where SOC values are being stored
+ * @param soc_stats soc_stats object to retrieve address from
+ * @return page being used to store SOC values
+ */
+uint16_t App_SocStats_GetSocAddress(struct SocStats *soc_stats);
+
+/**
+ * Return if the SOC value was corrupt on startup
+ * @param soc_stats soc_stats object to retrieve corrupt status from
+ * @return corrupt status
+ */
+bool App_SocStats_GetCorrupt(struct SocStats *soc_stats);
+
+/**
+ * Update the state of charge of all series elements using coulomb counting.
+ * @param soc_stats The charge stats of the pack
+ * @param current The current from the cell to be updated (- is out of the cell, + is into the cell)
+ */
+void App_SocStats_UpdateSocStats(struct SocStats *soc_stats, float current);
+
+/**
+ * return Coulombs of the SE with the lowest SOC
+ * @param soc_stats The charge stats of the pack
+ * @return cloulombs of lowest SOC SE
+ */
+float App_SocStats_GetMinSocCoulombs(struct SocStats *soc_stats);
+
+/**
+ * return percent SOC of the SE with the lowest SOC
+ * @param soc_stats The charge stats of the pack
+ * @return soc % of lowest SOC SE
+ */
+float App_SocStats_GetMinSocPercent(struct SocStats *soc_stats);
+
+/**
+ * Get the minimum series element open circuit voltage approximation given current pack SOC status
+ * @param soc_stats current SOC of each series element in pack
+ * @return float minimum series element open circuit voltage approximation
+ */
+float App_SOC_GetMinOcvFromSoc(struct SocStats *soc_stats);
```

Comment:
Instead of having a bunch of getters that are only sent over CAN, I like the pattern of just having a single "Broadcast" function that handles setting all the TX values. I did this on the accumulator in the BMS refactor and it cuts the file size in half, and is a lot clearer + more maintainable in my opinion

Reply:
Only `App_SocStats_GetMinSocPercent` is being broadcast over CAN. `App_SOC_GetMinOcvFromSoc` and `App_SocStats_GetMinSocCoulombs` are implemented for future use, but not currently utilized.

I can remove `App_SocStats_GetMinSocCoulombs` if it bothers you, but `App_SOC_GetMinOcvFromSoc` will be needed for current limiting in the near future.